agent/agent_endpoint.go

```diff
@@ -227,29 +234,68 @@ func (s *HTTPServer) AgentServices(resp http.ResponseWriter, req *http.Request)
 		return nil, err
 	}
 
-	services := s.agent.State.Services(&entMeta)
-	if err := s.agent.filterServicesWithAuthorizer(authz, &services); err != nil {
-		return nil, err
-	}
+	wch := make(chan struct{})
```

Comment:
The documentation states that state.Notify sends a message on the channel whenever a state change occur. It says that for now only service registration and deregistration, but that could change. I don't know if using this here is the best option.